### ISSUE TITLE
Cypress Component Testing runner crashes on webpack build error

### ISSUE THREAD

**Issue by yjaaidi** [cypress-io/cypress#16569]
### Current behavior

When using Component Testing + Webpack 5, build errors produce a blank screen in the Cypress runner and the following error in the browser console:
```
TypeError: input.join is not a function
    at Filter.toHtml (cypress_runner.js:17794)
    at cypress_runner.js:193178
    at cypress_runner.js:67030
    at cypress_runner.js:67008
    at trackDerivedFunction (cypress_runner.js:67195)
    at e.track (cypress_runner.js:67195)
    at useObserver (cypress_runner.js:67006)
    at wrappedComponent (cypress_runner.js:67030)
    at hg (cypress_runner.js:77416)
    at Mg (cypress_runner.js:77432)
    at Lg (cypress_runner.js:77432)
    at Tg (cypress_runner.js:77450)
    at bi (cypress_runner.js:77498)
    at ci (cypress_runner.js:77499)
    at Di (cypress_runner.js:77515)
    at Yh (cypress_runner.js:77514)
```

The problem seems to happen here:
https://github.com/cypress-io/cypress/blob/56234e52d6d1cbd292acdfd5f5d547f0c4706b51/packages/runner-ct/src/errors/script-error.jsx#L15

where with Webpack 5, the error is apparently not a string but an object with the string in the `message` property.

This should probably be replaced by something like:
```javascript
convert.toHtml(error.error?.message ?? error.error) 
```

### Desired behavior
It should not crash 😅.

### Test code to reproduce
1. Create an Angular project `ng new demo`
2. Setup cypress component testing by following these instructions https://github.com/jscutlery/test-utils/blob/main/packages/cypress-angular/README.md
3. Change the spec code in order to produce a compilation error.

### Versions

Cypress 7.3.0

**Comment by dgrbrady** [comment]
I'm also having issues with Angular component testing. I upgraded an app to Angular 12 (which now uses webpack 5), and get this error (the cypress runner never opens):

```
Error [TypeError]: Cannot set property plugins of #<Object> which has only a getter
    at Object.<anonymous> (/arsenal-web/node_modules/@cypress/webpack-dev-server/dist/makeWebpackConfig.js:80:39)
    at Generator.next (<anonymous>)
    at /arsenal-web/node_modules/@cypress/webpack-dev-server/dist/makeWebpackConfig.js:27:71
    at new Promise (<anonymous>)
    at __awaiter (/arsenal-web/node_modules/@cypress/webpack-dev-server/dist/makeWebpackConfig.js:23:12)
    at Object.makeWebpackConfig (/arsenal-web/node_modules/@cypress/webpack-dev-server/dist/makeWebpackConfig.js:45:12)
    at Object.<anonymous> (/arsenal-web/node_modules/@cypress/webpack-dev-server/dist/startServer.js:41:57)
    at Generator.next (<anonymous>)
    at /arsenal-web/node_modules/@cypress/webpack-dev-server/dist/startServer.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/arsenal-web/node_modules/@cypress/webpack-dev-server/dist/startServer.js:4:12)
    at Object.start (/arsenal-web/node_modules/@cypress/webpack-dev-server/dist/startServer.js:35:12)
    at /arsenal-web/node_modules/@cypress/webpack-dev-server/dist/index.js:18:54
    at Generator.next (<anonymous>)
    at /arsenal-web/node_modules/@cypress/webpack-dev-server/dist/index.js:8:71
    at new Promise (<anonymous>)
```

**Comment by lmiller1990** [comment]
I followed the instructions [here](https://github.com/jscutlery/test-utils/blob/main/packages/cypress-angular/README.md) but I keep getting:

```
Error: It is required to register dev-server plugin that implements `dev-server:start` event for component testing.
```

🤔

**Comment by yjaaidi** [comment]
The θmod issue means that Angular modules have not being precompiled with ngcc.

I just noticed an issue with our package where some Angular deps have been added automatically to the package.json while they should be peer deps. We end up with multiple versions of Angular core etc...

TL;DR: we released a fix. Can you try the 0.5.1?

Thx @lmiller1990 for investigating

**Comment by yjaaidi** [comment]
Thx for the quick fix @JessicaSachs @lmiller1990
